model/src/main/proto/math.proto

```diff
@@ -188,3 +188,105 @@ message MathEquation {
   // The MathExpression representing the left-hand side of the equation, e.g. the '0' in 2x+3y=0.
   MathExpression right_side = 2;
 }
+
+// Represents a list of comparable mathematics operations.
+//
+// 'Comparable' here means that this structure provides a trivial way to compare commutative and
+// associative operations (i.e. by extracting terms from multiple subsequent commutative &
+// associative operations into lists that can be deterministically sorted). This structure is meant
+// to provide a means to compare two expressions without considering associativity or commutativity
+// (though the latter requires the operation lists stored within this structure to be sorted before
+// using standard proto equals checking). Also note that care must be taken when performing equality
+// checking since this structure can contain floating point values that require an epsilon check to
+// approximate equality.
+message ComparableOperationList {
```

Comment:
As this message name says it is a list so either this message will be repeated somewhere else or inside this message we need a `repeated` value to make it a list!!

What I am understanding here is, this message is a list and list contains multiple items, so inside this message there should be a `repeated` field. 
```proto
repeated ComparableOperation root_operation = 1;
```

Reply:
I guess I think of this conceptually as a list even though it's a mix between a tree and a list. Do you have a suggestion for a clearer name?

@vinitamurthi as well in case you have any thoughts.

Reply:
When working on a later PR (that actually introduces the computations for this structure), I thought about this again & decided to remove the outer list wrapper proto (especially before I went too deep into the testing to save time later on). PTAL and let me know your thoughts.

Reply:
looks good to me. 
Keeping this conversation open for @vinitamurthi